DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java

```diff
@@ -0,0 +1,78 @@
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationOutputHandler;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.InvocationResult;
+import org.apache.maven.shared.invoker.Invoker;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+
+public class MavenInvokerBuilder {
+    private static final List<String> PUBLISH_GOALS = Arrays.asList("test", "install");
+    private StringBuilder output = new StringBuilder();
+    ;
+    /**
+     * Define a field for the calling program instance.
+     **/
+    private final Invoker invoker;
+
+    /**
+     * Instantiating the calling program in the class constructor
+     **/
+    public MavenInvokerBuilder(File localRepositoryDir) {
+        this.invoker = new DefaultInvoker();
+        this.invoker.setLocalRepositoryDirectory(localRepositoryDir);
```

Comment:
Does this work? Invoker is final so it can't be changed once initialized right?

Reply:
Yeah, it works. I think "final" just means this invoker won't be another invoker. But it is okay to set attributes for this one. Btw, tests were added.

Reply:
That makes sense